enemy/flying_eye/flying_eye.gd

```diff
@@ -41,10 +41,10 @@ func attack_state_enter():
 
 
 func death_state_enter():
-	SignalBus.enemy_died.emit(coin_reward)
 	died.emit()
 	anim.play("death")
 	collider.disabled = true
+	SignalBus.enemy_died.emit(coin_reward)
```

Comment:
no reason this needs to be at the top or bottom - was troubleshooting the problem with enemies continuing to attack after they were considered dead